### PR TITLE
Add tip on solving merge conflicts in functional tests

### PR DESCRIPTION
**Pull request by yucheng11122017**
**What is the purpose of this pull request?**

- [X] Documentation update
- [ ] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

**Overview of changes:**
Add tip on solving merge conflict in functional test expected results
Tip based on [comment](https://github.com/MarkBind/markbind/pull/2170#issuecomment-1472949660) by @tlylt 

**Anything you'd like to highlight/discuss:**

**Testing instructions:**

**Proposed commit message: (wrap lines at 72 characters)**
Add tip on solving merge conflicts in functional tests

---

**Checklist:** :ballot_box_with_check:

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [ ] Linked all related issues
- [X] No unrelated changes

### CODE REVIEW

File: docs/devGuide/development/workflow.md [hunk 1]
```diff
@@ -190,6 +190,20 @@ After which, you can update the **expected** test files with: `npm run updatetes
...
+<box type="tip" seamless header="Solving merge conflicts in expected test files">
+
+1. Ensure the fork is latest
+   - Pull from upstream repo to fork
+2. Ensure local master is in sync with fork master
+   - Pull from the fork master into local master
+3. Checkout to PR branch
+4. Merge local master into PR branch 
+   - `git merge master`
+5. Accept any change to conflicts in generated test files
```
Comment (itsyme) [inline docs/devGuide/development/workflow.md:202]: Perhaps we could include examples of files/filetypes of generated functional test files!

**Review by itsyme — APPROVED** [review]
Good work @yucheng11122017! LGTM besides the small nit! Useful tip that I think many future developers will find useful!

File: docs/devGuide/development/workflow.md (hunk 1 above)
Comment (yucheng11122017) [reply docs/devGuide/development/workflow.md:202]: Hi @itsyme, thanks for the comment! I added a tooltip explaining the location of these expected files above (`expected` for functional and `__snapshot__` for unit test)

File: docs/devGuide/development/workflow.md [hunk 2]
```diff
@@ -190,6 +190,20 @@ After which, you can update the **expected** test files with: `npm run updatetes
   as uncommitted changes due to the way they are generated. If you are not directly modifying those files in your PR, you should **discard those changes** and **do not commit** them.
 </box>
 
+<box type="tip" seamless header="Solving merge conflicts in expected test files">
+
+1. Ensure the fork is latest
```
Comment (lhw-1) [inline docs/devGuide/development/workflow.md:195]: Does "Ensure that the fork is synced with the upstream branch" work better?

File: docs/devGuide/development/workflow.md [hunk 3]
```diff
@@ -190,6 +190,20 @@ After which, you can update the **expected** test files with: `npm run updatetes
...
+   - Pull from upstream repo to fork
```
Comment (lhw-1) [inline docs/devGuide/development/workflow.md:196]: I think this should be fine to omit? As long as they know how to use GitHub, I think we can assume that they know how to sync a fork :thinking:

File: docs/devGuide/development/workflow.md [hunk 4]
```diff
@@ -190,6 +190,20 @@ After which, you can update the **expected** test files with: `npm run updatetes
...
+2. Ensure local master is in sync with fork master
+   - Pull from the fork master into local master
```
Comment (lhw-1) [inline docs/devGuide/development/workflow.md:198]: I think this overlaps with the first step!

File: docs/devGuide/development/workflow.md [hunk 5]
```diff
@@ -190,6 +190,20 @@ After which, you can update the **expected** test files with: `npm run updatetes
...
+3. Checkout to PR branch
```
Comment (lhw-1) [inline docs/devGuide/development/workflow.md:199]: We can include the command! `git checkout [BRANCH NAME]`

File: docs/devGuide/development/workflow.md [hunk 6]
```diff
@@ -190,6 +190,20 @@ After which, you can update the **expected** test files with: `npm run updatetes
...
+4. Merge local master into PR branch 
+   - `git merge master`
```
Comment (lhw-1) [inline docs/devGuide/development/workflow.md:201]: Perhaps we can specify that we are calling this command in the branch (and not the master branch)?

**Review by lhw-1 — COMMENTED** [review]
LGTM! Just some nits on the wording :+1:

File: docs/devGuide/development/workflow.md (hunk 2 above)
Comment (yucheng11122017) [reply docs/devGuide/development/workflow.md:195]: Thanks for the change! I think repo might be a better word than branch so its clearer :)

File: docs/devGuide/development/workflow.md (hunk 4 above)
Comment (yucheng11122017) [reply docs/devGuide/development/workflow.md:198]: I think what the first step is saying is remote fork is synced with upstream repo. Second step is saying that local master is in sync with remote fork :)

File: docs/devGuide/development/workflow.md (hunk 6 above)
Comment (yucheng11122017) [reply docs/devGuide/development/workflow.md:201]: I think this is ok to omit because after the third step, they should already be in the PR branch :)

**Comment by yucheng11122017** [comment]
Thanks for the review @lhw-1! Fixed the nits and responded to your comments :)

File: docs/devGuide/development/workflow.md [hunk 7]
```diff
@@ -190,6 +190,20 @@ After which, you can update the **expected** test files with: `npm run updatetes
   as uncommitted changes due to the way they are generated. If you are not directly modifying those files in your PR, you should **discard those changes** and **do not commit** them.
 </box>
 
+<box type="tip" seamless header="Solving merge conflicts in expected test files">
```
Comment (lhw-1) [inline docs/devGuide/development/workflow.md:193]:
```suggestion
<box type="tip" seamless header="Here are the steps to solve merge conflicts in expected test files:">
```

File: docs/devGuide/development/workflow.md [hunk 8]
```diff
@@ -190,6 +190,20 @@ After which, you can update the **expected** test files with: `npm run updatetes
...
+
+1. Ensure the fork is synced with the upstream repository
+2. Ensure local master is in sync with fork master
+   - Pull from the fork master into local master
+3. Checkout to PR branch
+   - `git checkout [BRANCH NAME]`
+4. Merge local master into PR branch 
+   - `git merge master`
+5. Accept any change to conflicts in generated test files
```
Comment (lhw-1) [inline docs/devGuide/development/workflow.md:202]:
```suggestion
5. Accept all changes to any merge conflicts in the generated **expected** test files.
```

File: docs/devGuide/development/workflow.md [hunk 9]
```diff
@@ -190,6 +190,20 @@ After which, you can update the **expected** test files with: `npm run updatetes
...
+   - It does not matter which change is accepted as it will be overridden in the following step
```
Comment (lhw-1) [inline docs/devGuide/development/workflow.md:203]:
```suggestion
   - It does not matter which changes are accepted, as they will be overridden in the following step.
```

File: docs/devGuide/development/workflow.md [hunk 10]
```diff
@@ -190,6 +190,20 @@ After which, you can update the **expected** test files with: `npm run updatetes
...
+6. Once the merge is done, run `npm run updatetest` to generate the latest test files
```
Comment (lhw-1) [inline docs/devGuide/development/workflow.md:204]:
```suggestion
6. Once your master branch has been successfully merged into your PR branch, run `npm run updatetest` to generate the latest test files.
```

**Review by lhw-1 — COMMENTED** [review]
Thanks for fixing the nits! Unfortunately, I have more nits (sorry) 😅

**Pull request merged**